File: 198-house-robber/198-house-robber.java
class Solution {
    public int rob(int[] choco) {
        int answer = 0;
    if(choco == null || choco.length == 0){
      return answer;
    }

    int [][] dp = new int [choco.length+1][2];

    for(int i = 1; i <= choco.length; i++){
      //if we choose the current chocolate
      dp[i][0] = choco[i-1] + dp[i-1][1];
      //if we dont choose, choose the max of previous step
      dp[i][1] = Math.max(dp[i-1][0],dp[i-1][1]);
    }

    return Math.max(dp[choco.length][0],dp[choco.length][1]);
    }
}

File: 198-house-robber/198-house-robber.py
from collections import defaultdict
class Solution(object):
    def rob(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        dp = defaultdict(int)
        def helper(ch,n):
            #print(ch)
            if ch >= n:
                return 0
            if ch in dp:
                return dp[ch]
            val = max(helper(ch+2,n) + nums[ch], helper(ch+1,n))
            dp[ch] = val
            return val
        
        return helper(0,len(nums))